jdbc/src/main/java/org/apache/zeppelin/jdbc/JDBCPoolManager.java

```diff
@@ -0,0 +1,135 @@
+import org.apache.zeppelin.tabledata.ColumnDef;
+import org.apache.zeppelin.tabledata.InterpreterResultTableData;
+import org.apache.zeppelin.tabledata.Row;
+import org.apache.zeppelin.tabledata.TableData;
+
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class JDBCPoolManager {
+  private static final String PARAGRAPH_ID_KEY = "paragraph_id=";
+  private static final String NOTE_ID_KEY = "note_id=";
+  // the table name must begin with a letter
+  private static final String SQL_NAME_PREFIX = "p";
+
+  private static String getPoolExpression(String sqlReq) {
+    final int poolReqBegIndex = sqlReq.indexOf("{ResourcePool");
+    final int poolReqEndIndex = sqlReq.indexOf("}", poolReqBegIndex);
+    return sqlReq.substring(poolReqBegIndex, poolReqEndIndex + 1);
+  }
+
+  private static String getParagraphId(String poolExp) {
+    final int pIdIndex = poolExp.indexOf(PARAGRAPH_ID_KEY);
+    return poolExp.substring(pIdIndex + PARAGRAPH_ID_KEY.length(), poolExp.length() - 1);
+  }
+
+  private static String getNoteId(String poolExp, InterpreterContext context) {
+    if (poolExp.contains(NOTE_ID_KEY)) {
+      final int nIdBegIndex = poolExp.indexOf(NOTE_ID_KEY);
+      final int nIdEndIndex = poolExp.indexOf(".", nIdBegIndex);
+      return poolExp.substring(nIdBegIndex + NOTE_ID_KEY.length(), nIdEndIndex);
+    } else {
+      return context.getNoteId();
+    }
+  }
+
+  private static TableData getTableDataFromResourcePool(ResourcePool resourcePool,
+                                                        String noteId, String paragraphId) {
+    final String str = "" + resourcePool.get(noteId, paragraphId,
+        WellKnownResourceName.ZeppelinTableResult.toString()).get();
+    InterpreterResultMessage msg = InterpreterResult.getMsgsFromString(str).get(0);
+    return new InterpreterResultTableData(msg);
+  }
+
+  private static List<String> getColumnNamesWithTypes(TableData tableData, String stringType) {
+    final ColumnDef[] columns = tableData.columns();
+    List<String> columDefs = new LinkedList<>();
+    for (final ColumnDef column : columns) {
+      columDefs.add(column.name() + " " + stringType);
+    }
+    return columDefs;
+  }
+
+  private static String getSqlDropTableReq(String tableName) {
+    return "DROP TABLE IF EXISTS " + tableName + ";";
+  }
+
+  private static String getSqlCreateTableReq(List<String> columDefs, String tableName) {
+    return "CREATE TABLE " + tableName + " (" + StringUtils.join(columDefs, ", ") + ");";
+  }
+
+  private static List<String> getSqlInsertReqs(TableData tableData, String tableName) {
+    List<String> reqs = new LinkedList<>();
+    for (Iterator<Row> iterator = tableData.rows(); iterator.hasNext(); ) {
+      Row row = iterator.next();
+      reqs.add("INSERT INTO " + tableName + " VALUES ("
```

Comment:
is this going to be SQL syntax independent?

Reply:
We rely on the [SQL standard](http://www.contrib.andrew.cmu.edu/~shadow/sql/sql1992.txt) from 1992. If database does not support this standard, this feature will not work.